step-capstone/src/components/TripList.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { Grid } from '@material-ui/core'
+import TripItemComponent from "./TripItemComponent"
+import Trip from "./Trip"
+import {
+  Switch,
+  Route,
+  Redirect
+} from "react-router-dom";
+
+class TripList extends React.Component{
+
+  constructor(props){
+    super(props);
+    this.state = {selectedTrip : null};
+    this.handleOpenTrip = this.handleOpenTrip.bind(this);
+  }
+
+  getTrips(userId) {
+    //Test Data
+    //TODO: Replace test data with call to firestore method
+    const tripTestOne = {
+      id : 1,
+      title : "London trip", 
+      startDate : new Date (2020, 8, 10),
+      endDate : new Date (2020, 8, 15),
+      description : "lorem ipsum ", 
+    }
+    const tripTestTwo = {
+      id : 2,
+      title : "Spain trip", 
+      startDate : new Date (2020, 12, 10),
+      endDate : new Date (2020, 12, 15),
+      description : "lorem ipsum ", 
+    }
+    return [tripTestOne, tripTestTwo];
+  }
+
+  handleOpenTrip(tripID) {
+    this.setState({ selectedTrip: tripID })
+  }
+
+  render () {
+    const trips = this.getTrips();
+    return (
+      <div>
+        <Switch>
+          <Route path = "/trips/:tripId">
```

Comment:
we probably don't want to reveal the tripID in the path since that's it's unique key in the database

Reply:
Good thought, but might not be a big deal for this application. If you're worried about it, you could use the trip name instead and ask for users to make unique names for their trips.

Reply:
I think it wouldnt matter. anyone who has managed to get access to the database would have access to the Ids. the Id itself doesnt allow anything i dont think